client/my-sites/stats/stats-module-utm/stats-module-utm.jsx

```diff
@@ -0,0 +1,168 @@
+import ErrorPanel from '../stats-error';
+import StatsListCard from '../stats-list/stats-list-card';
+import StatsModulePlaceholder from '../stats-module/placeholder';
+import UTMDropdown from './stats-module-utm-dropdown';
+import UTMExportButton from './utm-export-button';
+
+import '../stats-module/style.scss';
+import '../stats-list/style.scss';
+
+const OPTION_KEYS = {
+	SOURCE_MEDIUM: 'utm_source,utm_medium',
+	CAMPAIGN_SOURCE_MEDIUM: 'utm_campaign,utm_source,utm_medium',
+	SOURCE: 'utm_source',
+	MEDIUM: 'utm_medium',
+	CAMPAIGN: 'utm_campaign',
+};
+
+function generateFileNameForDownload( siteSlug, period ) {
+	// Build a filename for the CSV export button.
+	// The "format('L')" method can return strings that are not safe for the file system.
+	// While the "saveAs" function handles this, we do it here for correctness.
+	// We prefer '-' for word boundries, and '_' within words/dates.
+	// This allows text editing shortcuts to work properly.
+	//
+	// example output: jetpack.com-utm-day-03_20_2024-03_20_2024.csv
+	//
+	const newFileName =
+		[
+			siteSlug,
+			'utm',
+			period.period,
+			period.startOf.format( 'L' ),
+			period.endOf.format( 'L' ),
+		].join( '-' ) + '.csv';
+
+	return newFileName.replace( /\//g, '_' );
+}
+
+const StatsModuleUTM = ( {
+	path,
+	className,
+	useShortLabel,
+	moduleStrings,
+	summary,
+	period,
+	metricLabel,
+	hideSummaryLink,
+	isLoading,
+	query,
+	postId,
+} ) => {
+	const siteId = useSelector( getSelectedSiteId );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const translate = useTranslate();
+	const [ showLoader, setShowLoader ] = useState( isLoading );
+	const [ selectedOption, setSelectedOption ] = useState( OPTION_KEYS.SOURCE_MEDIUM );
+
+	const optionLabels = {
+		[ OPTION_KEYS.SOURCE_MEDIUM ]: {
+			selectLabel: translate( 'Source / Medium' ),
+			headerLabel: translate( 'Posts by Source / Medium' ),
+			isGrouped: true, // display in a group on top of the dropdown
+		},
+		[ OPTION_KEYS.CAMPAIGN_SOURCE_MEDIUM ]: {
+			selectLabel: translate( 'Campaign / Source / Medium' ),
+			headerLabel: translate( 'Posts by Campaign / Source / Medium' ),
+			isGrouped: true,
+		},
+		[ OPTION_KEYS.SOURCE ]: {
+			selectLabel: translate( 'Source' ),
+			headerLabel: translate( 'Posts by Source' ),
+		},
+		[ OPTION_KEYS.MEDIUM ]: {
+			selectLabel: translate( 'Medium' ),
+			headerLabel: translate( 'Posts by Medium' ),
+		},
+		[ OPTION_KEYS.CAMPAIGN ]: {
+			selectLabel: translate( 'Campaign' ),
+			headerLabel: translate( 'Posts by Campaign' ),
+		},
+	};
+
+	// Fetch UTM metrics with switched UTM parameters.
+	const { isFetching: isFetchingUTM, metrics: data } = useUTMMetricsQuery(
+		siteId,
+		selectedOption,
+		query,
+		postId
+	);
+
+	// Show error and loading based on the query
+	const hasError = false;
+
+	const displaySummaryLink = data && ! hideSummaryLink;
+
+	useEffect( () => {
+		setShowLoader( isLoading, isFetchingUTM );
```

Comment:
```suggestion
		const showLoader = isLoading || isFetchingUTM;
```

Could we use the check directly rather than controlling the state?

Reply:
@dognose24 yeah good question, I'm not sure why it was written in that way 🤔 

addressed in d3bedb9c336

Reply:
Previously the component was fetching data and accepting loading state from outside. Thanks for refactoring this.